packages/devextreme/testing/tests/DevExpress.core/utils.license.tests.js

```diff
@@ -0,0 +1,138 @@
+    const TOKEN_MISSING_FIELD_3 = 'ewogICJmb3JtYXQiOiAxLAogICJjdXN0b21lcklkIjogImIxMTQwYjQ2LWZkZTEtNDFiZC1hMjgwLTRkYjlmOGU3ZDliZCIKfQ==.resgTqmazrorRNw7mmtV31XQnmTSw0uLEArsmpzCjWMQJLocBfAjpFvKBf+SAG9q+1iOSFySj64Uv2xBVqHnyeNVBRbouOKOnAB8RpkKvN4sc5SDc8JAG5TkwPVSzK/VLBpQxpqbxlcrRfHwz9gXqQoPt4/ZVATn285iw3DW0CU=';
+    const TOKEN_UNSUPPORTED_VERSION = 'ewogICJmb3JtYXQiOiAyLAogICJjdXN0b21lcklkIjogImIxMTQwYjQ2LWZkZTEtNDFiZC1hMjgwLTRkYjlmOGU3ZDliZCIsCiAgIm1heFZlcnNpb25BbGxvd2VkIjogMjMxCn0=.tTBymZMROsYyMiP6ldXFqGurbzqjhSQIu/pjyEUJA3v/57VgToomYl7FVzBj1asgHpadvysyTUiX3nFvPxbp166L3+LB3Jybw9ueMnwePu5vQOO0krqKLBqRq+TqHKn7k76uYRbkCIo5UajNfzetHhlkin3dJf3x2K/fcwbPW5A=';
+
+    test('W0019 error should be logged if license is empty', function(assert) {
+        verifyLicense('', '1.0');
+        assert.equal(errors.log.callCount, 1);
+        assert.strictEqual(errors.log.getCall(0).args[0], 'W0019');
+
+        setLicenseCheckSkipCondition(false);
+
+        verifyLicense(null, '1.0');
+        assert.equal(errors.log.callCount, 2);
+        assert.strictEqual(errors.log.getCall(1).args[0], 'W0019');
+
+        setLicenseCheckSkipCondition(false);
+
+        verifyLicense(undefined, '1.0');
+        assert.equal(errors.log.callCount, 3);
+        assert.strictEqual(errors.log.getCall(2).args[0], 'W0019');
+    });
+
+    test('No messages should be logged if license is valid', function(assert) {
+        verifyLicense(TOKEN_23_1, '23.1');
+        assert.ok(errors.log.notCalled);
+
+        verifyLicense(TOKEN_23_1, '12.3');
+        assert.ok(errors.log.notCalled);
+
+        verifyLicense(TOKEN_23_2, '23.1');
+        assert.ok(errors.log.notCalled);
+
+        verifyLicense(TOKEN_23_2, '23.2');
+        assert.ok(errors.log.notCalled);
+    });
+
+    test('Message should be logged only once', function(assert) {
+        verifyLicense('', '1.0');
+        assert.ok(errors.log.calledOnce);
+
+        verifyLicense('', '1.0');
+        assert.ok(errors.log.calledOnce);
+    });
+
+    test('No messages should be logged if setLicenseCheckSkipCondition() used', function(assert) {
+        setLicenseCheckSkipCondition();
+        verifyLicense('', '1.0');
+        assert.ok(errors.log.notCalled);
+    });
+
+    test('W0020 error should be logged if license is outdated', function(assert) {
+        verifyLicense(TOKEN_23_1, '23.2');
+        assert.equal(errors.log.callCount, 1);
+        assert.strictEqual(errors.log.getCall(0).args[0], 'W0020');
+
+        setLicenseCheckSkipCondition(false);
+
+        verifyLicense(TOKEN_23_2, '42.4');
+        assert.equal(errors.log.callCount, 2);
+        assert.strictEqual(errors.log.getCall(1).args[0], 'W0020');
+    });
+
+    test('W0021 error should be logged if license is corrupted/invalid', function(assert) {
```

Comment:
We can rewrite this using forEach to get rid of copypaste of `setLicenseCheckSkipCondition`

```javascript
[
    [TOKEN_UNVERIFIED, '1.2.3']
].forEach((token, version) => {
    verifyLicense(token, version);
    assert.strictEqual(errors.log.lastCall.args[0], 'W0021');
    setLicenseCheckSkipCondition(false);
});
```